prime-router/src/main/kotlin/Metadata.kt

```diff
@@ -323,5 +357,19 @@ class Metadata {
         const val schemasSubdirectory = "schemas"
         const val valuesetsSubdirectory = "valuesets"
         const val tableSubdirectory = "tables"
+        const val fileNameTemplatesSubdirectory = "./file_name_templates"
+        @Volatile private var defaultMetadata: Metadata? = null
+        // I am probably threadsafe. If things go bananas verify I'm not the cause
+        // this is instead of doing everything with DI because doing DI right in Azure
+        // with existing DI libraries was extremely complex and beyond the scope
+        // of this work. And probably hard to get right. And honestly not necessary. Probably.
+        // honestly, the case could be made to make Metadata a singleton and then this code
+        // can go away, but that is also beyond the scope of this work right now
+        fun provideMetadata(): Metadata {
+            return defaultMetadata ?: synchronized(this) {
+                val newInstance = defaultMetadata ?: Metadata(defaultMetadataDirectory).also { defaultMetadata = it }
+                newInstance
+            }
+        }
```

Comment:
This does what the comment says it does: creates a threadsafe static method that lets us get the default Metadata object if we need it. In all honesty Metadata should probably be a singleton because its collections are essential to the operating of the system.

I have hammered against this a ton and haven't seen any issues with thread safety or race conditions or it being surprisingly null, BUT if we start to see problems, we should start here.

Reply:
Ask @RickHawesUSDS  - I think it was a singleton back in the early days, but Rick changed things to pass it in as an argument.